### PR TITLE
optimize/convex/lp: fix Simplex when number of constraints is number …

### PR DESCRIPTION
**Pull request by btracey**
…of variables

Fixes 441.

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @vladimir-ch @sbinet

File: optimize/convex/lp/simplex.go [hunk 1]
```diff
@@ -100,6 +100,19 @@ func simplex(initialBasic []int, c []float64, A mat.Matrix, b []float64, tol flo
 	}
 	m, n := A.Dims()
 
+	if m == n {
+		// Problem is exactly constrained, perform a linear solve.
+		bVec := mat.NewVecDense(len(b), b)
+		x := make([]float64, m)
```
Comment (vladimir-ch) [inline optimize/convex/lp/simplex.go]: `m == n` here but still it is strange to use `m` and `n` on the next line.

File: optimize/convex/lp/simplex.go [hunk 2]
```diff
@@ -100,6 +100,19 @@ func simplex(initialBasic []int, c []float64, A mat.Matrix, b []float64, tol flo
...
+		xVec := mat.NewVecDense(n, x)
+		err := xVec.SolveVec(A, bVec)
+		if err != nil {
+			return math.NaN(), nil, nil, ErrSingular
+		}
+		f := floats.Dot(x, c)
+		return f, x, nil, nil
```
Comment (vladimir-ch) [inline optimize/convex/lp/simplex.go]: What if not `x >= 0`?

File: optimize/convex/lp/simplex.go (hunk 1 above)
Comment (btracey) [reply optimize/convex/lp/simplex.go]: Done.

File: optimize/convex/lp/simplex.go (hunk 2 above)
Comment (btracey) [reply optimize/convex/lp/simplex.go]: Good point. Thanks.

**Pull request merged**